tests/report/host/tcbinfo.cpp

```diff
@@ -225,3 +261,245 @@ void TestVerifyTCBInfo(oe_enclave_t* enclave, const char* test_filename)
+
+    printf("TestVerifyTCBInfo V2: Positive Tests passed\n");
+
+    const char* negative_files[] = {
+        // In the following files, a property in corresponding level has been
+        // capitalized. JSON is case sensitive and therefore schema validation
+        // should fail.
+        "./data_v2/tcbInfoNegativePropertyMissingLevel0.json",
+        "./data_v2/tcbInfoNegativePropertyMissingLevel1.json",
+        "./data_v2/tcbInfoNegativePropertyMissingLevel2.json",
+        "./data_v2/tcbInfoNegativePropertyMissingLevel3.json",
+        // In the following files, a property in corresponding level has wrong
+        // type.
+        "./data_v2/tcbInfoNegativePropertyWrongTypeLevel0.json",
+        "./data_v2/tcbInfoNegativePropertyWrongTypeLevel1.json",
+        "./data_v2/tcbInfoNegativePropertyWrongTypeLevel2.json",
+        "./data_v2/tcbInfoNegativePropertyWrongTypeLevel3.json",
+
+        // Comp Svn greater than uint8_t
+        "./data_v2/tcbInfoNegativeCompSvn.json",
+
+        // pce Svn greater than uint16_t
+        "./data_v2/tcbInfoNegativePceSvn.json",
+
+        // Invalid issueDate field.
+        "./data_v2/tcbInfoNegativeInvalidIssueDate.json",
+
+        // Invalid nextUpdate field.
+        "./data_v2/tcbInfoNegativeInvalidNextUpdate.json",
+
+        // Missing nextUpdate field.
+        "./data_v2/tcbInfoNegativeMissingNextUpdate.json",
+
+        // Signature != 64 bytes
+        "./data_v2/tcbInfoNegativeSignature.json",
+
+        // Unsupported JSON constructs
+        "./data_v2/tcbInfoNegativeStringEscape.json",
+        "./data_v2/tcbInfoNegativeIntegerOverflow.json",
+        "./data_v2/tcbInfoNegativeIntegerWithSign.json",
+        "./data_v2/tcbInfoNegativeFloat.json",
+    };
+
+    for (size_t i = 0; i < sizeof(negative_files) / sizeof(negative_files[0]);
+         ++i)
+    {
+        std::vector<uint8_t> tcbInfo;
+        OE_TEST(FileToBytes(negative_files[i], &tcbInfo) == 0);
+
+        oe_parsed_tcb_info_t parsed_info = {0};
+        oe_tcb_tcb_level_t platform_tcb_level = {{0}};
+        oe_result_t ecall_result = OE_FAILURE;
+        OE_TEST(
+            test_verify_tcb_info(
+                enclave,
+                &ecall_result,
+                (const char*)&tcbInfo[0],
+                &platform_tcb_level,
+                &parsed_info) == OE_OK);
+        OE_TEST(ecall_result == OE_JSON_INFO_PARSE_ERROR);
+        printf(
+            "TestVerifyTCBInfoV2: Negative Test %s passed\n",
+            negative_files[i]);
+    }
+}
+
+void TestVerifyTCBInfoV2_AdvisoryIDs(
+    oe_enclave_t* enclave,
+    const char* test_filename)
+{
+    std::vector<uint8_t> tcbInfo;
+    oe_result_t ecall_result = OE_FAILURE;
+    OE_TEST(FileToBytes(test_filename, &tcbInfo) == 0);
+
+    oe_tcb_tcb_level_t platform_tcb_level = {
+        {4, 4, 2, 4, 1, 128, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}, 8};
+    oe_parsed_tcb_info_t parsed_info = {0};
+
+    // Set platform pce svn to 8 and assert that
```

Comment:
is pce_svn SGX-only or applicable across all TEE types?

Reply:
This is a SGX test.  There is an open issue #2205 to split out SGX attestation tests.